From 9f54743da85dbe667d85827e343a93c7ba6b0443 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 23 Nov 2023 16:29:30 +0100
Subject: [PATCH 0364/1695] dts: stm32: activate tamper node for
 STM32MP257F-EV1 board

Add tampers configuration for this board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c72bbfe3b6e3804ea82f7c4ae246f36158b67543 dts: stm32: activate tamper node for STM32MP257F-EV1 board]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I138ac01347a00319585afb6bd895c781c41cd201
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343127
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349411
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp257f-ev1.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index d5c73f4ad..6d6916d76 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -12,6 +12,7 @@
 #include <dt-bindings/soc/stm32mp25-risab.h>
 #include <dt-bindings/soc/stm32mp25-risaf.h>
 #include <dt-bindings/soc/stm32mp25-rifsc.h>
+#include <dt-bindings/tamper/st,stm32mp25-tamp.h>
 #include "stm32mp257.dtsi"
 #include "stm32mp25xf.dtsi"
 #include "stm32mp257f-ev1-ca35tdcid-rcc.dtsi"
@@ -58,6 +59,20 @@
 	status = "okay";
 };
 
+&tamp {
+	status = "okay";
+	st,tamp-passive-precharge = <2>;
+	st,tamp-passive-nb-sample = <4>;
+	st,tamp-passive-sample-clk-div = <16384>;
+
+	/* Tamper button */
+	tamp_button: tamp-passive@1 {
+		status = "disabled";
+		reg = <EXT_TAMPER_1 0>;
+		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+	};
+};
+
 &usart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.47.2

